secrets/drivertest/drivertest.go

```diff
@@ -28,8 +29,8 @@ import (
 // Harness descibes the functionality test harnesses must provide to run
 // conformance tests.
 type Harness interface {
-	// MakeDriver returns a driver.Keeper for use in tests.
-	MakeDriver(ctx context.Context) (driver.Keeper, error)
+	// MakeDriver returns a pair of driver.Keeper, each backed by a different key.
+	MakeDriver(ctx context.Context) (driver.Keeper, driver.Keeper, error)
```

Comment:
Do you think returning two keepers would be useful for more general cases? Otherwise how about calling makedriver multiple times when needed?

Reply:
The problem is that we need two separate keys, and `MakeDriver` would need to know which one to return. I did consider adding an argument to `MakeDriver` to choose which key, but it's awkward; e.g., would the argument be a boolean? An integer only allowed to be 0 or 1? A name that every provider test needs to check for? I ended up deciding that just returning two drivers was simpler. WDYT?

Reply:
Does random generating a key id each time affect replay tests? If not we can do that in MakeDriver, otherwise I think what you have now probably the simplest.

Reply:
We can't randomly generate a keyID for record/replay, the key has to be provisioned on the server. Even if we could, yes, it would affect the replay.

Another option would be to change the test to provision the key when the test starts up (for providers that need that), instead of creating it manually. That's what `pubsub` does. But that seems like a bigger change we could do separately if you think it's better.